lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
                 await this.#bot.equip(bestHarvestTool, 'hand');
                 return this.#bot.heldItem;
             } catch (err) {
-                console.error('Unable to equip a better tool', err);
+                this.#logWarn('Error trying to equip a better tool', err);
                 return null;
             }
         }
     }
 
+    /**
+     * Finds the best harvest tool in the bot inventory for mining the specified block.
+     * If we don't have the best tool, also checks if dig time is infinite because it can't be harvested without a tool
+     *
+     * @param block {Block}
+     * @returns {BestHarvestTool}
+     */
+    bestHarvestTool(block) {
+        const availableTools = this.#bot.inventory.items()
+        const effects = this.#bot.entity.effects
+
+        let fastest = Infinity
+        let bestTool = null
+        for (const tool of availableTools) {
+            const enchants = (tool && tool.nbt) ? nbt.simplify(tool.nbt).Enchantments : []
+            const digTime = block.digTime(tool ? tool.type : null, false, false, false, enchants, effects)
+            if (digTime < fastest) {
+                fastest = digTime
+                bestTool = tool
+            }
+        }
+
+        return new BestHarvestTool(bestTool, fastest)
+    }
+
+    /**
+     * <i><b>Experimental</b></i>
+     *
+     * This finds the most powerful melee attack item in the bot inventory
+     *
+     * Note: Today this only prioritizes weapon type, but does not prioritize weapon rarity/enchantments/etc
+     *
+     * @returns {Item|undefined}
```

Comment:
maybe null instead of undefined?

Reply:
This is definitely inconsistent throughout the file even before I got to it.  I'm fine with either in the doc, but `undefined` is the more technically accurate of the 2.

Reply:
I'm not sure if that is true, since JS defines functions with no return as returning undefined. I found that it is recommended that things that could be a value but are absent should be null. https://stackoverflow.com/a/37980662

I understand that it is undefined because the shift looks for index 0 which is undefined... but our result is defined, defined as the best melee item, and in this case we find that the item doesn't exist, and therefore null.